### DISCUSSION TITLE
Recombination Rate manipulation.

### DISCUSSION THREAD

**Discussion by joetabet** [gaynorr/AlphaSimR#46] [Q&A] [answered]
Hello, I am using AlphaSimR and simulating breeding programs. I am attempting to increase the recombination rate for each of my simulations. Is increasing the values of "genLen" and "bp" in the "runMacs2" function is the way to do it? Or there might be another approach to it I can use? And is there is a possibility of adding heritability for Recombination like in the previous version of AlphaSim?

Thank you

**Comment by gregorgorjanc** [comment, marked as answer]
@joetabet Yes, you should increase `genLen` to increase recombination rate.

I don't think you can manipulate recombination rate as a function of genetics in AlphaSimR at this stage, but @gaynorr can add more.

**Comment by janaobsteter** [comment]
A similar question, so I'll post it here. Is it possible to add a genetic map to the simulation of genomes with runMacs? Thanks!

**Comment by gregorgorjanc** [comment]
@janaobsteter we can do that in MaCS, so in principle it should be possible, but we need to check how to pass this between `runMacs2()` and underlying MaCS C++ code. Can you have a look?

**Comment by gaynorr** [comment]
@janaobsteter you could look at the code for `runMacs2` and how I use it to construct the manualCommand argument for `runMacs`. The `runMacs2` function is just a wrapper for `runMacs`. The catch is that you'd have to know how to construct the more advanced arguments for MaCS that allows you to vary recombination frequency across the chromosome. This isn't something that I could help with. It might be easier to use msprime or some other coalescent software program for this. 

In general, you don't specify a map for the coalescent programs. Instead, you specify how recombination rate varies across the chromosome and the map itself is determined by where mutations form. It is also worth considering if going to this extra trouble is really worth it in the end.
